lib/glimesh/api/resolvers/account_resolver.ex

```diff
@@ -29,6 +29,81 @@ defmodule Glimesh.Api.AccountResolver do
     end
   end
 
+  def follow_channel(
+        _parent,
+        %{channel_id: channel_id, live_notifications: live_notifications},
+        %{context: %{access: access}}
+      ) do
+    with :ok <- Bodyguard.permit(Glimesh.Api.Scopes, :follow, access),
```

Comment:
The way Elixr forces a single return from a method is interesting, had to read a bit about best patterns for handling failures. This is the cleanest approach I could find, any individual lookup can fail and the method will stop there and pass the error back to the caller, without throwing or having lots of nested if's.

The guards needed for the nil methods are kinda ugly. I think it would be cleaner if all the lookups returned ok/error result tuples, but that's a bigger change than I wanted to tackle. Also since all the error handling is the same it could potentially be extracted to a function. Or if needed, the individual lookups can be tagged so specialized errors can be generated for each like so:
```elixr
with :ok <- Bodyguard.permit(Glimesh.Api.Scopes, :follow, access),
     {:channel, channel} when channel != nil <- {:channel, Glimesh.ChannelLookups.get(channel_id, [:user])}
do
  # thing
else
  {:channel, nil} -> {:error, @error_no_such_channel}
end
```